Document posibility of unavailable guilds on Guild Create (#5122)
hammerandchisel/discord-api-docs: docs/topics/Gateway.md

This event can be sent in three different scenarios:

> 1.  When a user is initially connecting, to lazily load and backfill information for all unavailable guilds sent in the [Ready](#DOCS_TOPICS_GATEWAY/ready) event.

Reviewer: Removing the previous note just hampers understanding of the flow.

2.  When a Guild becomes available again to the client.

> 3.  When the current user joins a new Guild (this Guild may be available or unavailable).

Reviewer 2: ```suggestion
3.  When the current user joins a new Guild (this guild may be available or unavailable).
```

Reviewer 2: lowercase

Author: Guild seems to be used as upper case everywhere in this section, so for consistency sake will leave it like that

> 3.  When the current user joins a new Guild (this Guild may be available or unavailable).

Reviewer 3: ```suggestion
3.  When the current user joins a new Guild.

During an outage the guild object in scenarios 1 and 3 may be marked unavailable.
```

Reviewer: Yeah I prefer this wording to the one currently proposed.

The inner payload can be:

> - An available Guild: a [guild](#DOCS_RESOURCES_GUILD/guild-object) object with extra fields, as detailed bellow.

Reviewer 2: ```suggestion
- An available Guild: a [guild](#DOCS_RESOURCES_GUILD/guild-object) object with extra fields, as detailed below.
```

Reviewer 2: typo,

And maybe instead of `as detailed` something like `as written`, `as noted`?

- An unavailable Guild: an [unavailable guild](#DOCS_RESOURCES_GUILD/unavailable-guild-object) object.

> ###### Guild Create Available Guild Extra Fields

Reviewer: This shouldn't be renamed. It's already clear from the context that this is only available if the guild is available since that's the default state of a guild. Rewording this only makes it sound more awkward than it actually is.

> ###### Guild Create Available Guild Extra Fields

Reviewer 2: ```suggestion
###### Guild Create Extra Fields
```

Reviewer 2: I don't think that should be renamed

Author: These wont be sent for unavailable guilds, so its just to specify

Reviewer 3: "guild create available guild" is very difficult to read. 

```suggestion
###### Available Guild Create Extra Fields
```
